src/EditorFeatures/CSharpTest/Formatting/CSharpFormattingEngineTestBase.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Formatting;
+using Microsoft.CodeAnalysis.Test.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Formatting
+{
+    [UseExportProvider]
+    public class CSharpFormattingEngineTestBase : CoreFormatterTestsBase
+    {
+        protected override string GetLanguageName()
+            => LanguageNames.CSharp;
+
+        protected override SyntaxNode ParseCompilationUnit(string expected)
+            => SyntaxFactory.ParseCompilationUnit(expected);
+    }
+}
```

Comment:
note: the formatting test area needs a lot of TLC.  There is very little shared code, and there seem to be at least 5 *entirely different* ways to validate formatting, most of which call into our components at varying levels of the stack.  We really need to clean this up and unify it all.

This PR just attempts some smalls steps along that way.